draft-ietf-jsonpath-base.md

```diff
@@ -614,111 +608,99 @@ Queries:
 
 #### Examples
 {: unnumbered}
 
 JSON:
 
-    {"j": {"k": 3}}
+    {
+      "o": {"j": 1, "k": 2},
+      "a": [5, 3]
+    }
 
 Queries:
 
 | Query | Result | Result Paths | Comment |
 | :---: | ------ | :----------: | ------- |
-| `$.j`   | `{"k": 3}` | `$['j']`      | Named value of an object      |
-| `$.j.k` | `3`        | `$['j']['k']` | Named value in nested object  |
-{: title="Dot selector examples"}
+| `$.o[*]` | `1` <br> `2` | `$['o']['j']` <br> `$['o']['k']` | Object values      |
+| `$.o[*]` | `2` <br> `1` | `$['o']['k']` <br> `$['o']['j']` | Alternative result |
+| `$.a[*]` | `5` <br> `3` | `$['a'][0]` <br> `$['a'][1]`     | Array members      |
+{: title="Index wildcard selector examples"}
 
-### Dot Wildcard Selector {#wildcard}
+### Child Selector
 
 #### Syntax
-{: unnumbered}
+<!-- {: unnumbered} -->
 
-The dot wildcard selector has the form `.*` as defined in the
-following syntax:
+The child selector has the form `[<//PICKERS//>]` where `<//PICKERS//>` is a comma-delimited
+collection of one or more //PICKERS//.
+Each //PICKER// is defined below.
 
 ~~~~ abnf
-dot-wild-selector    = "." wildcard       ;  dot followed by asterisk
-wildcard             = "*"
+child-selector  = "[" S list-entry 1*(S "," S list-entry) S "]"
+
+list-entry     =  ( quoted-member-name /
+                    element-index      /
+                    slice-index /
+                    filter
+                  )
 ~~~~
 
 #### Semantics
-{: unnumbered}
+<!-- {: unnumbered} -->
 
-A `dot-wild-selector` acts as a wildcard by selecting the nodes of
-all member values of an object in its input nodelist as well as all
-element nodes of an array in its input nodelist.
-
-Applying the `dot-wild-selector` to a primitive JSON value (a number,
-a string, `true`, `false`, or `null`) selects no node.
-
-#### Examples
-{: unnumbered}
-
-JSON:
+A child selector operates on objects and arrays only.
+It contains a comma-delimited collection of //PICKERS// to indicate which
+object members and array elements to selects.
 
-    {
-      "o": {"j": 1, "k": 2},
-      "a": [5, 3]
-    }
+Each //PICKER// type functions differently and may be defined to operate on only
+objects, only arrays, or both.
 
-Queries:
+//PICKERS// of different types may be combined within a single value selector.
 
-| Query | Result | Result Paths | Comment |
-| :---: | ------ | :----------: | ------- |
-| `$.o.*` | `1` <br> `2` | `$['o']['j']` <br> `$['o']['k']` | Object values      |
-| `$.o.*` | `2` <br> `1` | `$['o']['k']` <br> `$['o']['j']` | Alternative result |
-| `$.a.*` | `5` <br> `3` | `$['a'][0]` <br> `$['a'][1]`     | Array members      |
-{: title="Dot wildcard selector examples"}
+The resulting nodelist of a child selector is the concatenation of
+the nodelists from each of its //PICKERS// in the order that the //PICKERS//
+appear in the list.
+Note that any node selected in more than one of the selector entries is kept
+as many times in the nodelist.
 
```

Comment:
This text confuses //PICKERS// and selectors. //PICKERS// don't produce nodelists unless they are used in a selector.

("the selector entries" is vestigial and needs to adopt similar language to the previous sentence, once that is ironed out.)

For me, the viability of this proposal hinges on two things:
* how cleanly/clearly the selector-//PICKERS// interface can be defined,
* how cleanly/clearly the general semantics of //PICKERS// can be defined (so far these semantics are only hinted),
* the readability of the resultant document structure for users (as well as implementers).

Reply:
I think "selector entries" is just meant to be "//PICKERS//" here.